pontoon/sync/tasks.py

```diff
@@ -172,100 +199,36 @@ def sync_sources(db_project, now, force, no_pull):
+    project_sync_log,
     now,
+    repos_changed,
+    repo_locales,
+    locales,
     added_paths=None,
     removed_paths=None,
     changed_paths=None,
     new_entities=None,
-    locale=None,
-    no_pull=False,
     no_commit=False,
-    full_scan=False,
+    force=False,
 ):
-    db_project = get_or_fail(
-        Project,
-        pk=project_pk,
-        message="Could not sync project with pk={0}, not found.".format(project_pk),
-    )
-
-    repos = db_project.translation_repositories()
-    repo_pk = repos[0].pk
-    repo = get_or_fail(
-        Repository,
-        pk=repo_pk,
-        message="Could not sync repo with pk={0}, not found.".format(repo_pk),
-    )
-
-    project_sync_log = get_or_fail(
-        ProjectSyncLog,
-        pk=project_sync_log_pk,
-        message=(
-            "Could not sync project {0}, log with pk={1} not found.".format(
-                db_project.slug, project_sync_log_pk
-            )
-        ),
-    )
+    repos = db_project.repositories.all()
+    repo = db_project.translation_repositories()[0]
 
     log.info("Syncing translations for project: {}".format(db_project.slug))
 
     repo_sync_log = RepositorySyncLog.objects.create(
         project_sync_log=project_sync_log, repository=repo, start_time=timezone.now()
     )
 
-    if locale:
-        locales = db_project.locales.filter(pk=locale.pk)
-    else:
-        locales = db_project.locales.all()
-
-    if not locales:
-        log.info(
-            "Skipping syncing translations for project {0}, no locales to sync "
-            "found within.".format(db_project.slug)
-        )
-        repo_sync_log.end()
-        return
-
-    # If project repositories have API access, we can retrieve latest commit hashes and detect
-    # changed locales before the expensive VCS pull/clone operations. When performing full scan,
-    # we still need to sync all locales.
-    if not full_scan:
-        locales = get_changed_locales(db_project, locales, now)
-
-    readonly_locales = db_project.locales.filter(project_locale__readonly=True)
     added_and_changed_resources = db_project.resources.filter(
         path__in=list(added_paths or []) + list(changed_paths or [])
     ).distinct()
 
-    # We should also sync files for which source file change - but only for read-only locales.
-    # See bug 1372151 for more details.
-    if added_and_changed_resources:
-        changed_locales_pks = [l.pk for l in locales]
-        readonly_locales_pks = [l.pk for l in readonly_locales]
-        locales = db_project.locales.filter(
-            pk__in=changed_locales_pks + readonly_locales_pks
-        )
-
-    # Pull VCS changes in case we're on a different worker than the one
-    # sync started on.
-    if not no_pull:
-        log.info("Pulling changes for project {0} started.".format(db_project.slug))
-        repos_changed, repo_locales = pull_changes(db_project, locales)
+    if repo_locales:
         repos = repos.filter(pk__in=repo_locales.keys())
```

Comment:
Looks like `repos` isn't used before line 362, so we can join both `if repo_locales:` into one down under.